app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java

```diff
@@ -254,49 +263,40 @@ private String getAddressSecondary(List<AddressComponents> addressComponentsList
   }
 
   private void fetchWeatherForecast(String latLng) {
-    getView().showProgressBar("");
 
-    // TODO refactor after mvp complete
-    Disposable disposable = weatherRepository.getForecast(latLng)
+    //TODO refactor after mvp complete
+    Disposable disposable = weatherRepositoryImpl.getForecast(latLng)
         .subscribeOn(Schedulers.io())
         .observeOn(AndroidSchedulers.mainThread())
-        .subscribeWith(new DisposableObserver<Forecast>() {
-          @Override public void onNext(Forecast forecast) {
+        .doOnSubscribe(d -> getView().showProgressBar(""))
+        .subscribeWith(new DisposableObserver<ForecastEntity>() {
+          @Override public void onNext(ForecastEntity forecast) {
             setForecast(forecast, latLng);
+            getView().hideProgressBar();
           }
 
           @Override public void onError(Throwable e) {
-            Timber.e(e);
-            getView().changeErrorVisibility(true);
-            getView().showErrorMessage();
+            Toast.makeText(getContext(), "Error fetching new data", Toast.LENGTH_SHORT).show();
+          /*  getView().changeErrorVisibility(true);
+            getView().showErrorMessage();*/
             getView().hideProgressBar();
           }
 
           @Override public void onComplete() {
-            getView().hideProgressBar();
           }
         });
-/*    forecast = weatherModel.fetchWeatherForecast(latLng);
-
-    setForecast(forecast, latLng);*/
-  }
-
-  public void onFetchError() {
-    getView().changeErrorVisibility(true);
-    getView().showErrorMessage();
-  }
 
-  public void onFetchComplete() {
-    getView().hideProgressBar();
+    disposables.add(disposable);
   }
 
-  private void setForecast(Forecast forecast, String latLng) {
-    currentForecast = forecast.getCurrentForecast();
-    dailyForecast = forecast.getDailyForecast();
-    dailyForecastList = dailyForecast.getDailyDataList();
-    hourlyForecast = forecast.getHourlyForecast();
-    hourlyDataList = hourlyForecast.getHourlyDataList();
-
+  private void setForecast(ForecastEntity forecast, String latLng) {
+    Timber.e("set forecast");
+    this.forecast = forecast;
+    currentForecast = forecast.getCurrentForecastEntity();
```

Comment:
if forecast object already contains all the variables, setting a new variable currentForecast is redundant. Remove this approach as it is hard to maintain